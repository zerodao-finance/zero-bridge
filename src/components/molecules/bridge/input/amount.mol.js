import { ConvertInput } from '../../../atoms/inputs'
import { FaBitcoin } from 'react-icons/fa'
import { library } from '../../../utils/tokens'
import { useEffect, useState } from 'react'
import { ethers, BigNumber } from 'ethers'
import _ from 'lodash'

export const Input = ({amount, effect, tokenPrice}) => {

    
    var formatter = new Intl.NumberFormat('en-US', {
        style: "currency",
        currency: "USD"
       })
    
    return (
       <div className="w-fit self-center px-0 py-0">
           <div className="w-fit flex items-center justify-between gap-10 dark:bg-gray-600 bg-gray-50 px-2 rounded-2xl">
               <div>
                   <p className="text-[10px] text-gray-500 whitespace-nowrap mb-1">FROM</p>
                   <span className="flex items-center gap-2">
                       <FaBitcoin className="w-max h-[2rem] fill-gray-400"/>
                       <p className="dark:text-white text-gray-500">
                       BTC
                        </p>
                   </span>
               </div>
               <ConvertInput value={amount} onChange={effect} />
           </div>
           <div className=" xl:mr-5 italic font-light tracking-wider w-full text-right text-[10px] text-emerald-500">
               ~ USD { tokenPrice && formatter.format(amount * ethers.utils.formatUnits(tokenPrice, 6)) }
           </div> 
       </div> 
    )
}
