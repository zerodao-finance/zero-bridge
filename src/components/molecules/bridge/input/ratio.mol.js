import { FaBtc, FaEthereum } from 'react-icons/fa'




export const Ratio = ({ ratio, effect }) => {
    return (
        
                <div className="flex flex-col gap-2 self-center text-black dark:text-white">
                    <div className="flex flex-row w-full justify-between items-center px-5 gap-2">
                        <span className="flex items-center"><FaEthereum className="fill-gray-900 dark:fill-gray-300 h-[1.3rem] w-max text-xs"/><p className="text-xs">{`${ratio}%`}</p></span>
                        <input type="range" className="z-40 overflow-hidden form-range appearance-none h-1 focus:outline-none focus:ring-0 focus:shadow-none rounded-full bg-slate-400 w-[15rem]" id="range_select" name="ratio" min="0" max="100" step="1" value={ratio} onChange={effect}/>
                        <span className="flex items-center font-bold"><p className="text-xs">{`${100-ratio}%`}</p><FaBtc className="fill-gray-900 dark:fill-gray-300 h-[1.3rem] w-max"/></span> 
                    </div>
                </div>
            
    )
}
