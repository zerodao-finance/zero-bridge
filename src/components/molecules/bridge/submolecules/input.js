import { ConvertInput } from '../../../atoms/inputs'
import { library } from '../../../utils/tokens'
import { useEffect, useState } from 'react'
import { ethers, BigNumber } from 'ethers'
import _ from 'lodash'

export const Input = ({_token, _context}) => {

    const [tokenPrice, setPrice] = useState(null)
    const token = library[_token]
    
    var formatter = new Intl.NumberFormat('en-US', {
        style: "currency",
        currency: "USD"
       })
    
    const {connection, connectWallet} = global.wallet
    useEffect(async () => {
       if (await token.priceFeed) setPrice(ethers.utils.formatUnits((await token.priceFeed).toString(), 4))
   }, [connection])
   const {state, dispatch} = _context() 
   
    return (
       <div className="w-fit self-center px-0 py-0">
           <div className="w-fit flex items-center justify-between gap-10 dark:bg-gray-600 bg-gray-50 px-2 rounded-2xl">
               <div>
                   <p className="text-[10px] text-gray-500 whitespace-nowrap mb-1">FROM</p>
                   <span className="flex items-center gap-2">
                       {token.icon}
                       <p className="dark:text-white text-gray-500">
                       {token.symbol}
                        </p>
                   </span>
               </div>
               <ConvertInput value={state.value} onChange={(e) => dispatch({type: "changeValue", event: e})} />
           </div>
           <div className=" xl:mr-5 italic font-light tracking-wider w-full text-right text-[10px] text-emerald-500">
               ~ USD { tokenPrice && formatter.format(state.value * tokenPrice)}
           </div> 
       </div> 
    )
}
