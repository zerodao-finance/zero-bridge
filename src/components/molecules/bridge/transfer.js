import { Input, Ratio, Result } from './submolecules'
import { AiOutlineArrowDown, AiOutlineClose } from 'react-icons/ai'

export const TransferTool = ({ _action, _isLoading, _context }) => {
    return (
        <>
            <div className={`animate-flip-in-hor-top [animation-delay:400ms] container h-max flex flex-row place-content-center w-[25rem] gap-5 justify-around pr-[4.5rem] items-center px-8 ${_isLoading && 'invisible'}`}>
                <p className="text-[10px] text-gray-300 whitespace-nowrap">transfer amount</p>
                <div className="flex flex-col">
                    {/* <Convert /> */}
                    <Input _token="bitcoin" _context={_context}/>
                </div>
            </div>
            <div className={`animate-flip-in-hor-top [animation-delay:500ms] ${_isLoading && 'invisible'} flex flex-col justify-center place-items-center mt-5`}>
                <Ratio _context={_context}/>
                <AiOutlineArrowDown className="mt-3 fill-black dark:fill-white"/>
            </div>
            <div className={` animate-flip-in-hor-top [animation-delay:600ms] container h-max flex flex-row  place-content-center w-[25rem] gap-5 justify-around pr-[4.5rem] items-center px-8 pt-10 ${_isLoading && 'invisible'} pb-4`}>
                <p className="text-[10px] text-gray-300 whitespace-nowrap">result</p>
                <div className="flex flex-col w-full">
                    <Result _context={_context}/>
                </div>
            </div>
            <button className={`animate-flip-in-hor-top [animation-delay:700ms] rounded-full bg-emerald-300 dark:bg-emerald-500 text-white px-3 py-2 mt-4 hover:scale-90 transition-all font-fine duration-150 hover:ring-2 ring-emerald-700 tracking-wider ${_isLoading && 'invisible'}`} onClick={_action}>
                Initiate & Sign
            </button>
        </>
    )
}

