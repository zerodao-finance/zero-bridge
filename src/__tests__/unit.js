import renderer from 'react-test-renderer';
import { BridgeTransferInput } from '../ui/molecules/bridge.transfer/bridge.transfer.input';
import { shallow } from 'enzyme';
import { ethers } from "ethers";

describe('Unit Testing for zeroBRIDGE', () => {
    // Utils
    const formatter = new Intl.NumberFormat('en-US', {
        style: "currency",
        currency: "USD"
    });

    // State
    const state = {
        inputValue: 0
    };

    // Rendererings
    const bridgeInput = renderer.create(
        <BridgeTransferInput value={state.inputValue} />
    );

    // Tests
    it('renders input correctly', () => {
        expect(bridgeInput).toMatchSnapshot();
    });

    it('should render input with state of 0', () => {
        const bridgeInputRoot = bridgeInput.root;
        expect(bridgeInputRoot._fiber.pendingProps.value).toBe(0);
    });

    it('should accept user input', () => {
        expect(state.inputValue).toEqual(0);

        const inputWrapper = shallow(<BridgeTransferInput value={state.inputValue} />);
        inputWrapper.simulate('change', state.inputValue = 0.1);

        expect(state.inputValue).toEqual(0.1);
    });

    it('should convert BTC input to USD', () => {
        const btcUsdPrice = 50000;
        const btcInputInUsd = formatter.format(
            state.inputValue * ethers.utils.formatUnits(btcUsdPrice, 6)
        );

        expect(btcInputInUsd).toBe("$5000.00");
    });
})