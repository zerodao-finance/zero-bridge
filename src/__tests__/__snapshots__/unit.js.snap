// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing for zeroBRIDGE renders input correctly 1`] = `
<div
  className="w-fit self-center px-0 py-0 scale-[0.8] md:scale-[1]"
>
  <div
    className="w-fit flex items-center justify-between gap-10 dark:bg-gray-600 bg-gray-50 px-2 rounded-2xl"
  >
    <div>
      <p
        className="text-[10px] text-gray-500 whitespace-nowrap mb-1"
      >
        FROM
      </p>
      <span
        className="flex items-center gap-2"
      >
        <svg
          className="w-max h-[2rem] fill-gray-400"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zm-141.651-35.33c4.937-32.999-20.191-50.739-54.55-62.573l11.146-44.702-27.213-6.781-10.851 43.524c-7.154-1.783-14.502-3.464-21.803-5.13l10.929-43.81-27.198-6.781-11.153 44.686c-5.922-1.349-11.735-2.682-17.377-4.084l.031-.14-37.53-9.37-7.239 29.062s20.191 4.627 19.765 4.913c11.022 2.751 13.014 10.044 12.68 15.825l-12.696 50.925c.76.194 1.744.473 2.829.907-.907-.225-1.876-.473-2.876-.713l-17.796 71.338c-1.349 3.348-4.767 8.37-12.471 6.464.271.395-19.78-4.937-19.78-4.937l-13.51 31.147 35.414 8.827c6.588 1.651 13.045 3.379 19.4 5.006l-11.262 45.213 27.182 6.781 11.153-44.733a1038.209 1038.209 0 0 0 21.687 5.627l-11.115 44.523 27.213 6.781 11.262-45.128c46.404 8.781 81.299 5.239 95.986-36.727 11.836-33.79-.589-53.281-25.004-65.991 17.78-4.098 31.174-15.792 34.747-39.949zm-62.177 87.179c-8.41 33.79-65.308 15.523-83.755 10.943l14.944-59.899c18.446 4.603 77.6 13.717 68.811 48.956zm8.417-87.667c-7.673 30.736-55.031 15.12-70.393 11.292l13.548-54.327c15.363 3.828 64.836 10.973 56.845 43.035z"
          />
        </svg>
        <p
          className="dark:text-white text-gray-500"
        >
          BTC
        </p>
      </span>
    </div>
    <div
      className="w-fit p-2"
    >
      <input
        className="dark:bg-gray-600 dark:text-white form-input text-center border-0 focus:ring-0 text-md font-medium text-gray-600 bg-transparent caret-green-500 z-40 w-fit max-w-[5rem] rounded-xl bg-gray-100"
        id="transfer-amount"
        type="number"
        value={0}
      />
    </div>
  </div>
  <div
    className=" xl:mr-5 italic font-light tracking-wider w-full text-right text-[10px] text-emerald-500"
  >
    ~ USD 
  </div>
</div>
`;
